src/self_encryptor.rs

```diff
@@ -1483,6 +1715,81 @@ mod tests {
         Ok(())
     }
 
+    async fn general_truncate_test(
+        old_size: u32,
+        new_size: u32,
+        num_chunks: u32,
+    ) -> Result<(), SelfEncryptionError> {
+        let bytes_len = old_size;
+        let mut rng = new_test_rng()?;
+        let bytes = random_bytes(&mut rng, bytes_len as usize);
+        let (data_map, storage) = {
+            let storage = SimpleStorage::new();
+            let se = SelfEncryptor::new(storage, DataMap::None)
+                .expect("First encryptor construction shouldn't fail.");
+            se.write(&bytes, 0)
+                .await
+                .expect("Writing to encryptor shouldn't fail.");
+            check_file_size(&se, bytes_len as u64).await;
+            se.close().await?
+        };
+
+        let (data_map2, storage) = {
+            // Start with an existing data_map.
+            let se = SelfEncryptor::new(storage, data_map)
+                .expect("Second encryptor construction shouldn't fail.");
+            se.truncate(new_size as u64)
+                .await
+                .expect("Truncating encryptor shouldn't fail.");
+            se.close().await?
```

Comment:
if put a `check_file_size` before calling the `close`, shall it be expected to be `new_size` ?

Reply:
Yes